### PR TITLE
Fixes #1428 Converts command arguments to environment variables and back again (merge to master)

### LINKED ISSUE #1428: Remove user credentials from Marathon command line
Problem:
Currently Marathon allows to add HTTP user credentials (ID & password) via --http-credentials command line parameter. When the process is running anyone can easily see these credentials in the "ps" command output.

Proposed solutions:
1. Modify --http-credentials parameter once the process is started so it does not display in the "ps" output.
2. Read credentials from a file (preferred). This way more than one user can be defined.
3. Authenticate users against LDAP (even better).

Thanks
Nikolay

### PR DESCRIPTION
**Pull request by brad-peters**
This change allows environment variables to be passing into marathon so they do not show up in ps.

### CODE REVIEW

**Comment by jdef** [comment]
ok to test

**Comment by jdef** [comment]
xref #4644

File: bin/marathon-framework [hunk 1]
```diff
@@ -92,9 +116,11 @@ function load_options_and_log {
         clean_cmd+=( "${i}" )
       fi
     done
-    "${clean_cmd[@]}"
+    cmd_to_env $clean_cmd
```
Comment (h0tbird) [inline bin/marathon-framework]: I think this should be `cmd_to_env "${clean_cmd[@]}"`

File: bin/marathon-framework [hunk 2]
```diff
@@ -92,9 +116,11 @@ function load_options_and_log {
...
+    run_jar
   else
-    logged marathon "${cmd[@]}" "$@"
+    cmd_to_env $cmd
```
Comment (h0tbird) [inline bin/marathon-framework]: I think this should be `cmd_to_env "${cmd[@]}"`

**Comment by h0tbird** [comment]
Fixed in https://github.com/mesosphere/marathon/commit/85a32314d6c60d620c1fa2cc8d5660b23190cf47

**Pull request merged**